qiskit/primitives/estimator.py

```diff
@@ -135,6 +136,13 @@ def _call(
                 sq_obs = (obs @ obs).simplify()
                 sq_exp_val = np.real_if_close(final_state.expectation_value(sq_obs))
                 variance = sq_exp_val - expectation_value**2
+                if variance < 0:
+                    warn(
+                        f"Negative variance encountered: {variance}. Variance is automatically "
+                        "treated as zero.",
+                        category=RuntimeWarning,
```

Comment:
I'm not sure which warning type is appropriate, `RuntimeWarning` and `UserWarning`. I see both are used in Terra. Is there any rule for Terra?
https://github.com/Qiskit/qiskit-terra/blob/86fb555871f4811a842380dbf0d54dcda34a4841/qiskit/compiler/transpiler.py#L302
https://github.com/Qiskit/qiskit-terra/blob/594ad019a33ed30725c7907242eda6a1c5d95a61/qiskit/compiler/assembler.py#L342

Reply:
I am not sure. It's not written in Contributing.md.

Reply:
I don't think we have clear criteria for `QiskitError` and `ValueError` as well.

Reply:
RuntimeWarning is "Base category for warnings about dubious runtime features."
And for example `np.sqrt(-1)` is `RuntimeWarning`.
So I think it's ok.

Reply:
Would it be nice to add some comment why this could be negative, e.g. "This is likely due to the finite number of shots." or so ?

Reply:
This should be caused by a numerical error or passing a non hermitian observable, not the finite effect of sampling.